backend/arm64/reg_class.ml

```diff
@@ -0,0 +1,265 @@
+[@@@ocaml.warning "+a-40-41-42"]
+
+module T = struct
+  type t =
+    | Int64
+    | Float128
+
+  let all =
+    [Int64; (* general purpose registers *) Float128 (* neon registers *)]
+
+  let first_available_register : t -> int = function
+    | Int64 -> 0
+    | Float128 -> 100
+
+  let num_available_registers : t -> int = function
+    | Int64 -> 23
+    | Float128 -> 32
+
+  let num_registers : t -> int = function Int64 -> 28 | Float128 -> 32
+
+  (* See "DWARF for the ARM 64-bit architecture (AArch64)" available from
+     developer.arm.com. *)
+
+  let int_dwarf_reg_numbers =
+    [| 0;
+       1;
+       2;
+       3;
+       4;
+       5;
+       6;
+       7;
+       8;
+       9;
+       10;
+       11;
+       12;
+       13;
+       14;
+       15;
+       19;
+       20;
+       21;
+       22;
+       23;
+       24;
+       25;
+       26;
+       27;
+       28;
+       16;
+       17
+    |]
+
+  let float_dwarf_reg_numbers =
+    [| 64;
+       65;
+       66;
+       67;
+       68;
+       69;
+       70;
+       71;
+       72;
+       73;
+       74;
+       75;
+       76;
+       77;
+       78;
+       79;
+       80;
+       81;
+       82;
+       83;
+       84;
+       85;
+       86;
+       87;
+       88;
+       89;
+       90;
+       91;
+       92;
+       93;
+       94;
+       95
+    |]
+
+  let dwarf_register_numbers reg_class =
+    match reg_class with
+    | Int64 -> int_dwarf_reg_numbers
+    | Float128 -> float_dwarf_reg_numbers
+
+  let int_reg_name =
+    [| "x0";
```

Comment:
Looks like the formatting of these arrays was lost in the move. It'd be nice ot keep the formatting for not, it helps understand the mapping of indexes.

Reply:
(locally disabled formatting in a couple of places.)